### PR TITLE
Share port support + example.

### PR DESCRIPTION
**Pull request by DisDis**
added more detailed logging.
added shared port support.
added example for multi isolate service.

### CODE REVIEW

**Comment by DisDis** [comment]
@sigurdm ping

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ pubspec.lock
 # Directory created by dartdoc
 doc/api/
 
+/.idea/
```
Comment (sigurdm) [inline .gitignore]: Let's add this in a separate PR.

File: example/isolate-service/bin/server.dart [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+  final server = Server([GreeterService()]);
+  await server.serve(address: address, port: port, shared: true);
+  _log.info('Service listening on port ${server.port}...');
+
+  return singleCallbackPort((SendPort resultPort) {
+    _log.info('Service shutdowning...');
+    sendFutureResult(Future.sync(server.shutdown), resultPort);
+  });
+}
```
Comment (sigurdm) [inline example/isolate-service/bin/server.dart]: Nit: Missing final newline

File: example/isolate-service/bin/server.dart [hunk 3]
```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2018, the gRPC project authors. Please see the AUTHORS file
```
Comment (sigurdm) [inline example/isolate-service/bin/server.dart]: nit: Should be 2019

File: example/isolate-service/bin/server.dart [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      print('[$isolateMarker] ${rec.level.name}:${rec.loggerName} ${rec.time}: ${rec.message} ${rec.error}');
+    } else {
+      print('[$isolateMarker] ${rec.level.name}:${rec.loggerName} ${rec.time}: ${rec.message}');
+    }
+  });
+}
+
+final Logger _log = new Logger('main');
+
+Future<void> main(List<String> args) async {
```
Comment (sigurdm) [inline example/isolate-service/bin/server.dart]: Could you turn this example into a test?
Probably you can follow the example of round_trip_test.dart to avoid sending protos.

File: example/isolate-service/bin/server.dart [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+class GreeterService extends GreeterServiceBase {
+  static final Logger _log = new Logger('GreeterService');
+  @override
+  Future<HelloReply> sayHello(ServiceCall call, HelloRequest request) async {
+    _log.info("sayHello name='${request.name}'");
+    return HelloReply()..message = 'Hello, ${request.name}!';
+  }
+}
+
+
```
Comment (sigurdm) [inline example/isolate-service/bin/server.dart]: Did you format this with dartfmt?

File: example/isolate-service/README.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
+# Description
+The hello world server and client demonstrate how to use Dart gRPC libraries to
```
Comment (sigurdm) [inline example/isolate-service/README.md]: This does not describe this example accurately.
But probably it would be better with a test than an example

File: lib/src/server/server.dart [hunk 7]
```diff
@@ -111,8 +116,8 @@ class Server {
       // timeout.
       connection.incomingStreams.listen((stream) {
         handler = serveStream_(stream);
-      }, onError: (error) {
-        print('Connection error: $error');
+      }, onError: (error, StackTrace stackTrace) {
+        _log.severe('Connection error: ', error, stackTrace);
```
Comment (sigurdm) [inline lib/src/server/server.dart]: While I don't disagree with this change, I am a bit worried about dragging in another dependency.
Can we do the logging part in another PR?

File: .gitignore (hunk 1 above)
Comment (DisDis) [reply .gitignore]: done, https://github.com/grpc/grpc-dart/pull/251

File: lib/src/server/server.dart (hunk 7 above)
Comment (DisDis) [reply lib/src/server/server.dart]: done https://github.com/grpc/grpc-dart/pull/252

**Comment by DisDis** [comment]
@sigurdm  I created 3 PR:
https://github.com/grpc/grpc-dart/pull/251
https://github.com/grpc/grpc-dart/pull/252
https://github.com/grpc/grpc-dart/pull/253
after that i will create example PR

File: example/isolate-service/bin/server.dart (hunk 4 above)
Comment (DisDis) [reply example/isolate-service/bin/server.dart]: I don't understand how to do the correct share port test. Any ideas?
